src/main/java/seedu/duke/command/DeleteNoteCommand.java

```diff
@@ -61,15 +63,23 @@ public String execute() {
                 isDeleted = notebook.deleteNote(title);
             }
 
-            if (isDeleted && title.isBlank()) {
+            if (isDeleted &&  title.isBlank()) {
+                StorageManager.deleteNoteContentFile(deletedTitle);
+                StorageManager.saveAllNoteDetails(notebook);
                 return Formatter.formatString(COMMAND_SUCCESSFUL_MESSAGE + deletedTitle);
             } else if (isDeleted) {
+                StorageManager.deleteNoteContentFile(title);
+                StorageManager.saveAllNoteDetails(notebook);
                 return Formatter.formatString(COMMAND_SUCCESSFUL_MESSAGE + title);
             } else {
                 return Formatter.formatString(COMMAND_UNSUCCESSFUL_MESSAGE);
             }
         } catch (IndexOutOfBoundsException exception) {
             return Formatter.formatString(COMMAND_UNSUCCESSFUL_MESSAGE);
+        } catch (IOException e) {
+            return Formatter.formatString("Unable to write to file");
```

Comment:
Put "unable to write file" as a COMMAND_UNSUCCESSFUL type variable. Then pass that variable instead